### PR TITLE
fix(xyk): add liquidity checks calculated amount instead of limit

### PR DESCRIPTION
**Pull request by Roznovjak**
Fixes: #553

### CODE REVIEW

**Comment by lumir-mrkva** [comment]
Task linked: [CU-2rx7dtm XYK - test correct asset_b amount in add_liquidity](https://app.clickup.com/t/2rx7dtm)

File: pallets/xyk/src/tests/liquidity.rs [hunk 1]
```diff
@@ -185,6 +185,34 @@ fn add_liquidity_as_another_user_should_work() {
 	});
 }
 
+#[test]
+fn add_liquidity_should_work_when_limit_exceeds_account_balance() {
```
Comment (enthusiastmartin) [inline pallets/xyk/src/tests/liquidity.rs:189]: this does not sound right.  if a limit is exceeded - stuff should not work. 

perhaps just better wording here. 

i would like to see a test where it fails.

Comment (Roznovjak) [reply pallets/xyk/src/tests/liquidity.rs]: Renamed.
The test where it fails is [here](https://github.com/galacticcouncil/Basilisk-node/blob/7a1bddd4670c7dcc4b14489b8b5799685f91246d/pallets/xyk/src/tests/liquidity.rs#L395)

File: pallets/xyk/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = 'pallet-xyk'
-version = '5.0.1'
+version = '5.0.2'
```
Comment (lumir-mrkva) [inline pallets/xyk/Cargo.toml:3]: 5.1.0

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]
please bump runtime version

**Review by enthusiastmartin — APPROVED** [review]

**Review by jak-pan — APPROVED** [review]

**Pull request merged**